tests/scripts/audit-validity-dates.py

```diff
@@ -0,0 +1,427 @@
+        test_dir = self.find_test_dir()
+        test_data_folder = os.path.join(test_dir, 'data_files')
+        data_files = []
+        for (dir_path, _, file_names) in os.walk(test_data_folder):
+            data_files.extend(os.path.join(dir_path, file_name)
+                              for file_name in file_names)
+        return data_files
+
+class FileWrapper():
+    """
+    This a stub class of generate_test_code.FileWrapper.
+
+    This class reads the whole file to memory before iterating
+    over the lines.
+    """
+
+    def __init__(self, file_name):
+        """
+        Read the file and initialize the line number to 0.
+
+        :param file_name: File path to open.
+        """
+        with open(file_name, 'rb') as f:
+            self.buf = f.read()
+        self.buf_len = len(self.buf)
+        self._line_no = 0
+        self._line_start = 0
+
+    def __iter__(self):
+        """Make the class iterable."""
+        return self
+
+    def __next__(self):
+        """
+        This method for returning a line of the file per iteration.
+
+        :return: Line read from file.
+        """
+        # If we reach the end of the file.
+        if not self._line_start < self.buf_len:
+            raise StopIteration
+
+        line_end = self.buf.find(b'\n', self._line_start) + 1
+        if line_end > 0:
+            # Find the first LF as the end of the new line.
+            line = self.buf[self._line_start:line_end]
+            self._line_start = line_end
+            self._line_no += 1
+        else:
+            # No LF found. We are at the last line without LF.
+            line = self.buf[self._line_start:]
+            self._line_start = self.buf_len
+            self._line_no += 1
+
+        # Convert byte array to string with correct encoding and
+        # strip any whitespaces added in the decoding process.
+        return line.decode(sys.getdefaultencoding()).rstrip() + '\n'
+
+    def get_line_no(self):
+        """
+        Gives current line number.
+        """
+        return self._line_no
+
+    line_no = property(get_line_no)
+
+class SuiteDataAuditor(Auditor):
+    """Class for auditing files in tests/suites/*.data"""
+    def __init__(self, options):
+        super().__init__(options)
+        self.default_files = self.collect_default_files()
+
+    def collect_default_files(self):
+        """Collect all files in tests/suites/*.data"""
+        test_dir = self.find_test_dir()
+        suites_data_folder = os.path.join(test_dir, 'suites')
+        data_files = glob.glob(os.path.join(suites_data_folder, '*.data'))
+        return data_files
+
+    def parse_file(self, filename: str):
+        """
+        Parse a list of AuditData from file.
+
+        :param filename: name of the file to parse.
+        :return list of AuditData parsed from the file.
+        """
+        audit_data_list = []
+        data_f = FileWrapper(filename)
+        for _, _, _, test_args in parse_suite_data(data_f):
```

Comment:
Since the format of `.data` files is fairly simple, we don't need very precise parsing, and the classes and methods in `generate_test_code` are not very convenient, maybe it would be better to parse the `.data` file independently?

Just consider only lines that match `\A\w+:(.*:)?"`. That skips comments, dependency lines and all existing description lines. In theory a description line could contain `:"` but in practice it doesn't happen and is very unlikely to happen.

Reply:
> Just consider only lines that match \A\w+:(.*:)?". That skips comments, dependency lines and all existing description lines. 

I hope I'm not making mistake, but the pattern would capture the dependency lines.

Anyway, I'm not quite understand why we need a seperate parser for `.data` file when, from my point of view, `generate_test_code.parse_test_data` works perfectly.

Reply:
> the pattern would capture the dependency lines
No: those don't contain a `"`.

> generate_test_code.parse_test_data works perfectly.

I wouldn't say that. For example, [it doesn't complain about a misspelled `depends_on`](https://github.com/Mbed-TLS/mbedtls/pull/7441#pullrequestreview-1396003685).

Reply:
> No: those don't contain a `"`.

You are right, I missed the `"`.

> [it doesn't complain about a misspelled depends_on.](https://github.com/Mbed-TLS/mbedtls/pull/7441#pullrequestreview-1396003685)

I see, let's use a seperate parser.